### PR TITLE
Cleanup content-type protobuf constants

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

**Comment by caesarxuchao** [comment]
Can you add a release note about the breaking change in the go signature?

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go [hunk 1]
```diff
@@ -71,20 +71,18 @@ func IsNotMarshalable(err error) bool {
 // as-is (any type info passed with the object will be used).
 //
 // This encoding scheme is experimental, and is subject to change at any time.
-func NewSerializer(creater runtime.ObjectCreater, typer runtime.ObjectTyper, defaultContentType string) *Serializer {
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go]: @smarterclayton is there a reason for this to vary (e.g. streaming), or is this safe to hard code?

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go]: This isn't vary in the current codebase (we have also RawSerializer for protobuf for streaming purpose) - I don't see any reason for having that.
But I'm happy to hear @smarterclayton opinion too.

Comment (smarterclayton) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go]: The content type used in Unknown is always content protobuf.  So this should be safe if it's going into Unknown only.

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf_extension.go [hunk 2]
```diff
@@ -21,20 +21,12 @@ import (
...
 
-const (
-	// contentTypeProtobuf is the protobuf type exposed for Kubernetes. It is private to prevent others from
-	// depending on it unintentionally.
-	// TODO: potentially move to pkg/api (since it's part of the Kube public API) and pass it in to the
-	//   CodecFactory on initialization.
-	contentTypeProtobuf = "application/vnd.kubernetes.protobuf"
-)
-
 func protobufSerializer(scheme *runtime.Scheme) (serializerType, bool) {
```
Comment (smarterclayton) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf_extension.go]: Wait, I removed this file.  Oh wait, it hasn't merged yet.  Please cherry-pick my change in #76054 that removes this file.

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf_extension.go]: Done

**Pull request merged**